File: samples/java/app/controllers/Application.java
package controllers;

import play.*;
import play.cache.Cache;
import play.mvc.*;

import views.html.*;

public class Application extends Controller {
  
  public static Result index() {
    return ok(index.render("Your new application is ready."));
  }

    public static Result set() {
        Cache.set("key", "cached value");
        return ok("Cached.");
    }

    public static Result get() {
        return ok("Cached value = " + Cache.get("key"));
    }

    public static Result delete() {
        Cache.set("key", "foooo", 0);
        return ok("Deleted.");
    }

    public static Result cacheInt() {
        Cache.set("key", new Integer(123));

        Object cachedValue = Cache.get("key");

        if (cachedValue != null) {
            Integer integerValue = (Integer)cachedValue;
            return ok(cachedValue.toString() + cachedValue.getClass() + " " + integerValue + " " + integerValue.getClass());
            // 123 class java.lang.Integer 123 int
        } else {
            throw new RuntimeException("Could not get the cached value.");
        }
    }

    public static Result cacheString() {
        Cache.set("key", "mikoto");

        return ok(Cache.get("key").toString());
        // Some(mikoto)
    }

    public static Result cacheBool() {
        Cache.set("key", new Boolean(true));

        Object cachedValue = Cache.get("key");

        if (cachedValue != null) {
            Boolean booleanValue = (Boolean)cachedValue;
            return ok(cachedValue.toString() + " : " + cachedValue.getClass() + ", " + booleanValue + " : " + booleanValue.getClass());
            // true : class java.lang.Boolean, true : boolean
        } else {
            throw new RuntimeException("Could not get the cached value.");
        }
    }

}

File: samples/scala/app/controllers/Application.scala
package controllers

import play.api._
import play.api.mvc._
import play.api.Play.current
import cache.Cache

object Application extends Controller {
  

  def index = Action {
    Ok(views.html.index("Your new application is ready."))
  }
  
  def set = Action {
    Cache.set("key", "cached value")
    Ok("Cached.")
  }
  
  def get = Action {
    Ok(Cache.get("key").toString)
  }
  
  def delete = Action {
    Cache.set("key", "foooo", 0)
    Ok("Deleted.")
  }

  def cacheInt = Action {
    Cache.set("key", 123)

    Cache.get("key") match {
      case Some(cachedValue) =>
        val intValue = cachedValue.asInstanceOf[Int]
        Ok(cachedValue.toString + cachedValue.getClass + " " + intValue + " " + intValue.getClass)
        // 123 class java.lang.Integer 123 int
    }
  }
  
  def cacheString = Action {
    Cache.set("key", "mikoto")
    
    Ok(Cache.get("key").toString)
    // Some(mikoto)
  }
  
  def cacheBool = Action {
    Cache.set("key", true)
    
    Cache.get("key") match {
      case Some(cachedValue) =>
        val boolValue = cachedValue.asInstanceOf[Boolean]
        Ok(cachedValue.toString + " : " + cachedValue.getClass + ", " + boolValue + " : " + boolValue.getClass)
        // true : class java.lang.Boolean, true : boolean
    }
  }
  
}
